compiler/src/iree/compiler/DispatchCreation/FuseHorizontalContractions.cpp

```diff
@@ -283,336 +220,247 @@ static std::optional<HorizontalFusionGroup> getHorizontalFusionGroupMembers(
+    return std::nullopt;
   }
 
-  SmallVector<Value> rhsValues;
-  SmallVector<Value> initValues;
-  for (auto op : fusionGroup.contractionOps) {
-    Value rhs = op.getDpsInputOperand(1)->get();
-    Value init = op.getDpsInitOperand(0)->get();
-    rhsValues.push_back(rhs);
-    initValues.push_back(init);
-  }
-  Value newContractRhs = concatenateValues(rewriter, loc, rhsValues);
-  Value newContractInit = concatenateValues(rewriter, loc, initValues);
-
-  auto baseContractResultType =
-      cast<RankedTensorType>(baseContractOp->getResult(0).getType());
-  SmallVector<int64_t> newContractResultShape = {
-      static_cast<int64_t>(rhsValues.size())};
-  newContractResultShape.append(baseContractResultType.getShape().begin(),
-                                baseContractResultType.getShape().end());
-  auto newContractResultType = RankedTensorType::get(
-      newContractResultShape, baseContractResultType.getElementType());
-
-  Value lhs = baseContractOp->getOperand(0);
-
-  SmallVector<utils::IteratorType> newContractIteratorTypes = {
-      utils::IteratorType::parallel};
-  newContractIteratorTypes.append(baseContractOp.getIteratorTypesArray());
-
-  SmallVector<AffineMap> newContractIndexingMaps =
-      baseContractOp.getIndexingMapsArray();
-  newContractIndexingMaps[0] = getConcatenatedIndexingMap(
-      rewriter, newContractIndexingMaps[0], /*shiftOnly=*/true);
-  newContractIndexingMaps[1] =
-      getConcatenatedIndexingMap(rewriter, newContractIndexingMaps[1]);
-  newContractIndexingMaps[2] =
-      getConcatenatedIndexingMap(rewriter, newContractIndexingMaps[2]);
-
-  linalg::GenericOp newContractOp = rewriter.create<linalg::GenericOp>(
-      loc, newContractResultType, ValueRange{lhs, newContractRhs},
-      newContractInit, newContractIndexingMaps, newContractIteratorTypes);
-  rewriter.cloneRegionBefore(baseContractOp->getRegion(0),
-                             newContractOp.getRegion(),
-                             newContractOp.getRegion().begin());
-
-  linalg::LinalgOp concatResultOp = newContractOp;
-  if (fusionGroup.truncateOps) {
-    SmallVector<Value> newTruncOperands;
-    SmallVector<AffineMap> newTruncIndexingMaps;
-    linalg::GenericOp baseTruncOp = fusionGroup.truncateOps->front();
-    SmallVector<AffineMap> baseTruncOpIndexingMaps =
-        baseTruncOp.getIndexingMapsArray();
-
-    rewriter.setInsertionPoint(baseTruncOp);
-    if (failed(moveOperandDefs(
-            rewriter,
-            ArrayRef<linalg::GenericOp>(fusionGroup.truncateOps.value()),
-            baseTruncOp, dominanceInfo, fusionGroup.contractionOps))) {
-      return baseTruncOp.emitOpError(
-          "failed to move operand defs for truncate operations");
+  SmallVector<Value> fusedIns;
+  SmallVector<Value> fusedOuts;
+  SmallVector<Type> fusedResultTypes;
+  SmallVector<AffineMap> fusedInsIndexingMaps;
+  SmallVector<AffineMap> fusedOutsIndexingMaps;
+
+  linalg::LinalgOp seedOp = cast<linalg::LinalgOp>(linalgOps.front());
```

Comment:
style nit: use `auto` because `cast` already spells the type.

https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable